pkg/proxy/dialer.go

```diff
@@ -34,31 +35,46 @@ func ciliumDialer(ctx context.Context, network, address string) (net.Conn, error
 
 	fd, err := syscall.Socket(family, syscall.SOCK_STREAM, 0)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to create socket: %s", err)
 	}
 
 	f := os.NewFile(uintptr(fd), addr.String())
 	defer f.Close()
 
 	c, err := net.FileConn(f)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to create FileConn: %s", err)
 	}
 
-	if id, ok := identityFromContext(ctx); ok {
-		setSocketMark(c, id)
+	if identity != 0 {
+		setSocketMark(c, identity)
 	}
 
 	sockAddr, err := ipToSockaddr(family, addr.IP, addr.Port, addr.Zone)
 	if err != nil {
 		c.Close()
-		return nil, err
+		return nil, fmt.Errorf("unable to create sockaddr: %s", err)
+	}
+
+	if err := syscall.SetNonblock(fd, false); err != nil {
+		c.Close()
+		return nil, fmt.Errorf("unable to put socket in blocking mode: %s", err)
```

Comment:
nit : blocking --> non-blocking ?

Reply:
It is changing the socket from non-blocking to blocking. The error is correct.